src/Symfony/Component/HttpClient/Internal/CurlClientState.php

```diff
@@ -65,8 +65,9 @@ public function __construct(int $maxHostConnections, int $maxPendingPushes)
             return;
         }
 
-        curl_multi_setopt($this->handle, \CURLMOPT_PUSHFUNCTION, function ($parent, $pushed, array $requestHeaders) use ($maxPendingPushes) {
-            return $this->handlePush($parent, $pushed, $requestHeaders, $maxPendingPushes);
+        // Keep a dummy "onPush" reference to work around a refcount bug in PHP
+        curl_multi_setopt($this->handle, \CURLMOPT_PUSHFUNCTION, $this->onPush = static function ($parent, $pushed, array $requestHeaders) use ($maxPendingPushes) {
```

Comment:
the closure and the method cannot be made static, since the method still references `$this`

Reply:
oh, please also add a `private $onPush` property